Assets/Packages/com.studio23.ss2.splashscreensystem/Runtime/UI/SplashScreenUIManager.cs

```diff
@@ -0,0 +1,118 @@
+using Cysharp.Threading.Tasks;
+using Studio23.SS2.SplashScreenSystem.Core;
+using Studio23.SS2.SplashScreenSystem.Data;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Studio23.SS2.SplashScreenSystem.UI
+{
+    public class SplashScreenUIManager : MonoBehaviour
+    {
+        public static SplashScreenUIManager Instance;
+        public GameObject ButtonPanel;
+
+        public GameObject ParentPanel;
+        public TextMeshProUGUI titleText;
+        public GameObject ScrollRect;
+        public Transform rectParent;
+
+        public GameObject imagePrefab;
+        private ScriptableObject _splashData;
+        private List<GameObject> imageObjects = new List<GameObject>();
+
+        public SplashScreenBehaviour SplashScreenBehaviour;
+
+        //Button
+        public Button AcceptBtn;
+        public Button DeclineBtn;
+
+        private void Awake()
+        {
+            if (Instance == null)
+                Instance = this;
+            else
+                Destroy(gameObject);
+        }
+
+
+        [ContextMenu("Populate UI")]
+        void PopulateUI()
+        {
+            ThirdPartyData thirdPartyData = _splashData as ThirdPartyData;
+
+            foreach (var entry in thirdPartyData.ThirdPartyEntries)
+            {
+                GameObject newImage = Instantiate(imagePrefab, rectParent);
+                newImage.GetComponent<Image>().sprite = Sprite.Create(entry.image, new Rect(0, 0, entry.image.width, entry.image.height), Vector2.zero);
+                imageObjects.Add(newImage);
+                rectParent.GetComponent<GridLayoutGroup>().cellSize = new Vector2(entry.image.width, entry.image.height);
+            }
+        }
+
+        public async void CrossFadeData(float duration)
+        {
+            float val = 0.0f;
+            ParentPanel.GetComponent<CanvasGroup>().alpha = val;
+            while (val < duration)
+            {
+                val += Time.deltaTime;
+                var ratio = val / duration;
+                ParentPanel.GetComponent<CanvasGroup>().alpha = ratio;
```

Comment:
You are using getcomponent() every frame. get it once and cache it into a variable